web/packages/teleport/src/Apps/AddApp/useAddApp.ts

```diff
@@ -48,12 +52,24 @@ export default function useAddApp(ctx: TeleportContext) {
     }
   }, [automatic]);
 
+  async function fetchJoinToken() {
+    const req: JoinTokenRequest = { roles: ['App'], suggestedLabels: labels };
+    let resp: JoinToken;
+    try {
+      resp = await ctx.joinTokenService.fetchJoinTokenV2(req);
+    } catch (err) {
```

Comment:
Doing a quick drive by to say that this probably should try the fallback only on some specific error and not all errors – I suppose some variation of "not implemented" error? Otherwise the UI will always try v1 if v2 fails, even if it failes for an unrelated reason (e.g. network error).

Reply:
i'm confused... i do do that though? https://github.com/gravitational/teleport/pull/51058/files/dab143f6e054865b5bebd3b51f386cd257841590#diff-fa0e6e600754b82989be39d78e0ce1958319113f6ed394dab7b3eb3680e67fffR81

or am i missing something?

Reply:
Oh apologies, I didn't notice that the check is hidden within `tryV1Fallback`.